sensor/common/scannerdefinitions/http_handler.go

```diff
@@ -20,29 +22,48 @@ var (
 	headersToProxy = set.NewFrozenStringSet("If-Modified-Since", "Accept-Encoding")
 )
 
-// scannerDefinitionsHandler handles requests to retrieve scanner definitions
+// ScannerDefinitionHandler handles requests to retrieve scanner definitions
 // from Central.
-type scannerDefinitionsHandler struct {
-	centralClient *http.Client
+type ScannerDefinitionHandler struct {
+	centralClient    *http.Client
+	centralReachable *atomic.Bool
 }
 
 // NewDefinitionsHandler creates a new scanner definitions handler.
-func NewDefinitionsHandler(centralEndpoint string) (http.Handler, error) {
+func NewDefinitionsHandler(centralEndpoint string) (*ScannerDefinitionHandler, error) {
 	client, err := clientconn.NewHTTPClient(mtls.CentralSubject, centralEndpoint, 0)
 	if err != nil {
 		return nil, errors.Wrap(err, "instantiating central HTTP transport")
 	}
-	return &scannerDefinitionsHandler{
-		centralClient: client,
+	return &ScannerDefinitionHandler{
+		centralClient:    client,
+		centralReachable: &atomic.Bool{},
 	}, nil
 }
 
-func (h *scannerDefinitionsHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+// Notify reacts to sensor going into online/offline mode.
+func (h *ScannerDefinitionHandler) Notify(e common.SensorComponentEvent) {
+	switch e {
+	case common.SensorComponentEventCentralReachable:
+		h.centralReachable.Store(true)
+	case common.SensorComponentEventOfflineMode:
+		h.centralReachable.Store(false)
+	}
```

Comment:
nit: I think convention is to add `default` even if it's blank

Reply:
Done